drivers/serial/uart_cc23x0.c

```diff
@@ -0,0 +1,407 @@
+#endif
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.fifo_fill = uart_cc23x0_fifo_fill,
+	.fifo_read = uart_cc23x0_fifo_read,
+	.irq_tx_enable = uart_cc23x0_irq_tx_enable,
+	.irq_tx_disable = uart_cc23x0_irq_tx_disable,
+	.irq_tx_ready = uart_cc23x0_irq_tx_ready,
+	.irq_rx_enable = uart_cc23x0_irq_rx_enable,
+	.irq_rx_disable = uart_cc23x0_irq_rx_disable,
+	.irq_tx_complete = uart_cc23x0_irq_tx_complete,
+	.irq_rx_ready = uart_cc23x0_irq_rx_ready,
+	.irq_err_enable = uart_cc23x0_irq_err_enable,
+	.irq_err_disable = uart_cc23x0_irq_err_disable,
+	.irq_is_pending = uart_cc23x0_irq_is_pending,
+	.irq_update = uart_cc23x0_irq_update,
+	.irq_callback_set = uart_cc23x0_irq_callback_set,
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+#define UART_CC23X0_IRQ_CFG(n)                                                                     \
+	do {                                                                                       \
+		const struct uart_cc23x0_config *config = dev->config;                             \
+                                                                                                   \
+		UARTClearInt(config->reg, UART_INT_RX);                                            \
+		UARTClearInt(config->reg, UART_INT_RT);                                            \
+                                                                                                   \
+		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority), uart_cc23x0_isr,            \
+			    DEVICE_DT_INST_GET(n), 0);                                             \
+		irq_enable(DT_INST_IRQN(n));                                                       \
+	} while (false)
+
+#define UART_CC23X0_INT_FIELDS .callback = NULL, .user_data = NULL,
+#else
+#define UART_CC23X0_IRQ_CFG(n)
+#define UART_CC23X0_INT_FIELDS
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+#define UART_CC23X0_DEVICE_DEFINE(n)                                                               \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(n, uart_cc23x0_init_##n, NULL, &uart_cc23x0_data_##n,                \
+			      &uart_cc23x0_config_##n, PRE_KERNEL_1, CONFIG_SERIAL_INIT_PRIORITY,  \
+			      &uart_cc23x0_driver_api)
+
+#define UART_CC23X0_INIT_FUNC(n)                                                                   \
+	static int uart_cc23x0_init_##n(const struct device *dev)                                  \
+	{                                                                                          \
+		struct uart_cc23x0_data *data = dev->data;                                         \
+		int ret;                                                                           \
+                                                                                                   \
+		CLKCTLEnable(CLKCTL_BASE, CLKCTL_UART0);                                           \
```

Comment:
please, capture clocks in device tree.

Reply:
Actually, CLKCTL_BASE is the base address for clock control registers. It's not a frequency for some particular clock.

Knowing that, does your comment still apply ? If so, I can eventually specify some `reg` property but AFAICS, this is not specified in most dtsi.

Reply:
@gmarull I updated the PR. I took into account your comments except this one. It's not being ignored, but the discussion is still pending: if I finally need to take care of it, of course I will. For now I'm not sure (these CLKCTL registers do not specifically refer to a particular clock, they are just used to enable/disable each peripheral's clock).

Thank you for your feedbacks (and for your presentations about Zephyr topics online, these videos were very useful for me)